File: stars-app/backend/models/alert.py
from pydantic import BaseModel


class AlertItem(BaseModel):
    alert_id: str
    severity: str
    title: str
    body: str
    meta: str | None = None
    measure_code: str | None = None
    cta_label: str | None = None
    cta_page: str | None = None


class PriorityItem(BaseModel):
    measure_code: str
    measure_name: str
    current_rate: float
    target_rate: float
    gap: float
    priority_score: int
    owner: str


class AlertsResponse(BaseModel):
    alerts: list[AlertItem]
    priority_board: list[PriorityItem]

File: stars-app/frontend/src/types/alert.ts
export interface AlertItem {
  alert_id: string;
  severity: "critical" | "warning" | "info";
  title: string;
  body: string;
  meta?: string | null;
  measure_code?: string | null;
  cta_label?: string | null;
  cta_page?: string | null;
}

export interface PriorityItem {
  measure_code: string;
  measure_name: string;
  current_rate: number;
  target_rate: number;
  gap: number;
  priority_score: number;
  owner: string;
}

export interface AlertsResponse {
  alerts: AlertItem[];
  priority_board: PriorityItem[];
}
